src/interface/swarm_container.hpp

```diff
@@ -0,0 +1,167 @@
+//========================================================================================
+// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef INTERFACE_SWARM_CONTAINER_HPP_
+#define INTERFACE_SWARM_CONTAINER_HPP_
+
+#include "globals.hpp"
+#include "swarm.hpp"
+#include <map>
+#include <memory>
+#include <string>
+#include <utility> // <pair>
+#include <vector>
+
+namespace parthenon {
+///
+/// Interface to underlying infrastructure for particle data declaration and
+/// access.
+/// Date: August 22, 2019
+///
+///
+/// The SwarmContainer class is a container for the swarms of particles that
+/// make up the simulation.
+///
+/// The container class will provide the following methods:
+///
+
+class MeshBlock;
+
+class SwarmContainer {
+ public:
+  //-----------------
+  // Public Variables
+  //-----------------
+  MeshBlock *pmy_block = nullptr; // ptr to MeshBlock
+
+  //-----------------
+  // Public Methods
+  //-----------------
+  // Constructor does nothing
+  SwarmContainer() {}
+
+  ///
+  /// Set the pointer to the mesh block for this container
+  void setBlock(MeshBlock *pmb) { pmy_block = pmb; }
```

Comment:
This method makes no sense because the `pmy_block` is a public variable so you are not enforcing encapsulation. It would be safer to move the variable to private and use methods to access it (setter/getter). This set method can be declared noexcept as it is.

Reply:
I think this is an artifact of what was going on in `Container` before I cleaned up the pointers. @brryan probably copied that machinery. @brryan take a look at what's going on in container in `develop` for the new preferred way of doing this.

Reply:
Yes this was just the state of the code when I copied it over. I'll update.